module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Driver/AbstractMultiDriverTest.php

```diff
@@ -0,0 +1,422 @@
+            'configpath'
+        );
+
+        $driver->getStatus('d1.123');
+    }
+
+    /**
+     *  Tests that logging works correctly
+     *
+     * @return array
+     */
+    public function testLogging()
+    {
+        $logger = new \Laminas\Log\Logger();
+        $writer = new \Laminas\Log\Writer\Mock();
+        $logger->addWriter($writer);
+
+        $driver = $this->initDriver(
+            [
+                'configLoader' => $this->getMockFailingConfigPluginManager(new RuntimeException()),
+            ]
+        );
+        $driver->setLogger($logger);
+
+        $this->callMethod($driver, 'getDriverConfig', ['bad']);
+        $this->assertEquals(
+            $driver::class . ': Could not load config for bad',
+            $writer->events[0]['message']
+        );
+
+        return ['driver' => $driver, 'writer' => $writer];
+    }
+
+    /**
+     * Test that MultiBackend can properly retrieve a new driver.
+     *
+     * @return void
+     */
+    public function testGetDriver()
+    {
+        //Set up the mock driver to be retrieved
+        $ILS = $this->getMockILS('Voyager', ['init', 'setConfig']);
+        $ILS->expects($this->once())
+            ->method('init');
+        $ILS->expects($this->once())
+            ->method('setConfig')
+            ->with(['config' => 'values']);
+
+        //Set up the ServiceLocator so it returns our mock driver
+        $driver = $this->initDriver(
+            ['driverManager' => $this->getMockSM($this->once(), 'Voyager', $ILS)]
+        );
+
+        //Add an entry for our test driver to the array of drivers
+        $drivers = ['testing2' => 'Voyager'];
+        $this->setProperty($driver, 'drivers', $drivers);
+
+        $returnDriver = $this->callMethod($driver, 'getDriver', ['testing2']);
+        $this->assertEquals($ILS, $returnDriver);
+
+        $returnDriver = $this->callMethod($driver, 'getDriver', ['nonexistent']);
+        $this->assertNull($returnDriver);
+    }
+
+    /**
+     *  Tests that getDriverConfig works correctly
+     *
+     * @return void
+     */
+    public function testGetDriverConfig()
+    {
+        $configData = ['config' => 'values'];
+        $driver = $this->initDriver();
+        $val = $this->callMethod($driver, 'getDriverConfig', ['good']);
+        $this->assertEquals($configData, $val);
+
+        $driver = $this->initDriver(
+            ['configLoader' => $this->getMockFailingConfigPluginManager(new RuntimeException())]
+        );
+        $val = $this->callMethod($driver, 'getDriverConfig', ['bad']);
+        $this->assertEquals([], $val);
+    }
+
+    /**
+     * Get a mock ILS authenticator
+     *
+     * @param string $userSource Source id, if the authenticator should emulate a
+     * situation where a user has logged in. Set to null for the attempt to cause an
+     * exception.
+     *
+     * @return \VuFind\Auth\ILSAuthenticator
+     */
+    protected function getMockILSAuthenticator($userSource = '')
```

Comment:
This method is only relevant in MultiBackendTest (at least for now). Is it worth moving, or do you prefer to keep it in the abstract base class?

Reply:
done